src/main/java/com/minecolonies/coremod/colony/buildings/BuildingGuardTower.java

```diff
@@ -503,16 +498,35 @@ public void setAssignManually(final boolean assignManually)
     }
 
     /**
+     * Possible job/AI.
+     */
+    public enum GuardJob
+    {
+        KNIGHT,
+        RANGER,
+    }
+
+    /**
+     * Possible tasks.
+     */
+    public enum Task
+    {
+        FOLLOW,
+        GUARD,
+        PATROL
+    }
+
+    /**
      * The client view for the baker building.
      */
     public static class View extends AbstractBuildingWorker.View
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this class from 56 to the maximum allowed 50 or externalize it in a public class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2972)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2176)